docs/extend/manifest.md

```diff
@@ -0,0 +1,239 @@
+<!--[metadata]>
++++
+aliases = [
+"/engine/extend/"
+]
+title = "Plugin manifest"
+description = "How develop and use a plugin with the managed plugin system"
+keywords = ["API, Usage, plugins, documentation, developer"]
+advisory = "experimental"
+[menu.main]
+parent = "engine_extend"
+weight=1
++++
+<![end-metadata]-->
+
+# Plugin manifest version 0
+
+This document outlines the format of of the V0 plugin manifest. The plugin
+manifest described herein was introduced in the Docker daemon (experimental version) in the [v1.12.0
+release](https://github.com/docker/docker/commit/f37117045c5398fd3dca8016ea8ca0cb47e7312b).
+
+Plugin manifests describe the various constituents of a docker plugin. Plugin
+manifests can be serialized to JSON format with the following media types:
+
+Manifest Type  | Media Type
+------------- | -------------
+manifest  | "application/vnd.docker.distribution.plugin.v0+json"
+
+
+## *Manifest* Field Descriptions
+
+Manifest provides the base accessible fields for working with V2 image format
+ in the registry.
+
+- **`manifestVersion`** *string*
+
+	version of the plugin manifest (vX.X)
+
+- **`description`** *string*
+
+	description of the plugin
+
+- **`documentation`** *string*
+
+  	link to the documentation about the plugin
+
+- **`interface`** *PluginInterface*
+
+   interface implemented by the plugins, struct consisting of the following fields
+      
+    - **`types`** *string array*
+
+      types indicate what interface(s) the plugin currently implements.
+      
+    - **`socket`** *string*
+      
+      socket is the name of the socket the engine should use to communicate with the plugins.
+
+
+- **`entrypoint`** *string array*
+
+   entrypoint of the plugin
```

Comment:
Lets have more description here. Feel free to use content from https://docs.docker.com/engine/reference/builder/#/entrypoint.